NewsApp/app/src/main/java/com/example/newsapp/data/room/NewsDao.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.newsapp.data.room
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.example.newsapp.data.entities.NewsEntity
+
+@Dao
+interface NewsDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertNews(newsEntity: NewsEntity): Long
```

Comment:
According to this [link ](https://commonsware.com/AndroidArch/pages/chap-roomconflict-001) I can't see this implementation will replace the entities that are duplicated because we relaying on auto-generated primary keys.

Reply:
the primary key is a must, right? which field should we rely on for the primary key?